src/components/Notification.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react'
+import { useParams, Link, useLocation } from 'react-router-dom'
+import { useQueryParameters } from './App'
+import { List, Label } from 'semantic-ui-react'
+
+export const NotificationsList: React.FC = () => {
+  const queryParameters = useQueryParameters()
+
+  return (
+    <div>
+      <h1>Notification Center</h1>
+      <p>List of notifications for the current user.</p>
+      <p>Can be filtered with query parameters, e.g:</p>
+
+      <List>
+        <List.Item>
+          <Link to="?status=unread">Unread Notifications</Link>
+        </List.Item>
+        <List.Item>
+          <Link to="?days=30&appid=1234">
+            Notifications in the last 30 days associated with Application ID: 1234
+          </Link>
+        </List.Item>
+        <List.Item>
+          <Link to={useLocation().pathname}>Reset query</Link>
+        </List.Item>
+      </List>
+      {Object.keys(queryParameters).length > 0 && <h4>Query parameters:</h4>}
+      <List>
+        {Object.entries(queryParameters).map(([key, value]) => (
+          <List.Item>{key + ' : ' + value}</List.Item>
+        ))}
+      </List>
+      <h4>Notifications:</h4>
+      <Label>
+        <Link to={'/notifications/567'}>Your message — click to view</Link>
+      </Label>
+    </div>
+  )
+}
+
+type TParams = { notificationId: string }
+
+export const Notification: React.FC = () => {
+  const { notificationId }: TParams = useParams()
+
+  return (
+    <div>
+      <h1>Title of Message</h1>
+      <p>Dear user, this message has the id: {notificationId}.</p>
+      <p>
+        <Link to="./">Back to Notification Center</Link>
+      </p>
+    </div>
+  )
+}
```

Comment:
I know this is just an example, but we should really break down into separated files. For the future. Do you agree?

Reply:
Not sure. I only put these into one file because I was getting tired of making so many small files and having a massive `import` list. I agree that it's good to keep things well contained, but I think when you have a few small(ish) components that all work together, then it's fine to keep them in one file. Especially if they share common functions or types, then it saves having to keep importing things all over the place. 

Seems to me the more isolated you make each component, the more messy and convoluted it becomes to keep track of all the imports.

Happy to go with whatever we decide though.

Reply:
I don't think this should be defined (or changed in this PR) what I was suggesting is that we create the habit of creating new files for every new component. Since they will grow in size most definitely in the future. What is shared between then should be passed as props or added to the context area, this way there wouldn't be much else to share. For now I understand that was just not worth, but let's work with that in mind for the next issues?